src/workerd/api/web-socket.c++

```diff
@@ -529,7 +529,6 @@ void WebSocket::startReadLoop(jsg::Lock& js, kj::Maybe<kj::Own<InputGate::Critic
       } else {
         native.closedIncoming = true;
         reportError(js, kj::cp(e));
-        kj::throwFatalException(kj::mv(e));
```

Comment:
This throw doesn't have much purpose in a context waitUntil task outside of failing the invocation and causing an exception trace on an otherwise successful request.

The error can and should be caught by the user worker with 
```js
ws.addEventListener("error", e => {
		...
      });
```